addons/mrp/migrations/11.0.2.0/post-migration.py

```diff
@@ -120,10 +120,20 @@ def create_stock_move_lines_from_stock_move_lots(env):
             sm.restrict_partner_id,
             sm.picking_id,
             sm.product_id,
-            0,
-            sm.product_uom,
-            0,
-            sm.product_uom_qty as qty_done,
+            CASE WHEN sm.is_done = TRUE THEN 0
+                 WHEN sml.lot_id IS NOT NULL THEN sml.quantity
+            ELSE 0
+            END as product_qty,
+            CASE WHEN sml.lot_id IS NOT NULL THEN spl.product_uom_id
+            ELSE sm.product_uom
+            END as product_uom_id,
+            CASE WHEN sm.is_done = TRUE THEN 0
+                 WHEN sml.lot_id IS NOT NULL THEN sml.quantity
+            ELSE 0
+            END as product_uom_qty,
+            CASE WHEN sml.lot_id IS NOT NULL THEN sml.quantity_done
```

Comment:
Do something similar as proposed above

Reply:
but here cannot. Here there is only one WHEN hahaha